spec/develop-spec.js

```diff
@@ -1,84 +1,75 @@
-    spyOn(Develop.prototype, "getRepositoryUrl").andCallFake((packageName, callback) => callback("Here is the error"));
-
-    const callback = jasmine.createSpy('callback');
-    apm.run(['develop', "fake-package"], callback);
-
-    waitsFor('waiting for develop to complete', () => callback.callCount === 1);
-
-    return runs(function() {
-      expect(callback.mostRecentCall.args[0]).toBe("Here is the error");
-      expect(fs.existsSync(repoPath)).toBeFalsy();
-      return expect(fs.existsSync(linkedRepoPath)).toBeFalsy();
-    });
-  }));
-
-  describe("when the repository hasn't been cloned", () => it("clones the repository to ATOM_REPOS_HOME and links it to ATOM_HOME/dev/packages", function() {
-    const Develop = require('../lib/develop');
-    spyOn(Develop.prototype, "getRepositoryUrl").andCallFake(function(packageName, callback) {
-      const repoUrl = path.join(__dirname, 'fixtures', 'repo.git');
-      return callback(null, repoUrl);
+  describe("when the package doesn't have a published repository url", () => {
+    it('logs an error', () => {
+      const Develop = require('../lib/develop');
+      spyOn(Develop.prototype, 'getRepositoryUrl').andCallFake((packageName, callback) => {
+        callback('Here is the error');
+      });
+      const callback = jasmine.createSpy('callback');
+      apm.run(['develop', 'fake-package'], callback);
+      waitsFor('waiting for develop to complete', () => callback.callCount === 1);
+      runs(() => {
+        expect(callback.mostRecentCall.args[0]).toBe('Here is the error');
+        expect(fs.existsSync(repoPath)).toBeFalsy();
+        expect(fs.existsSync(linkedRepoPath)).toBeFalsy();
+      });
     });
-    spyOn(Develop.prototype, "installDependencies").andCallFake(function(packageDirectory, options) {
-      return this.linkPackage(packageDirectory, options);
-    });
-
-    const callback = jasmine.createSpy('callback');
-    apm.run(['develop', "fake-package"], callback);
-
-    waitsFor('waiting for develop to complete', () => callback.callCount === 1);
+  });
 
-    return runs(function() {
-      expect(callback.mostRecentCall.args[0]).toBeFalsy();
-      expect(fs.existsSync(repoPath)).toBeTruthy();
-      expect(fs.existsSync(path.join(repoPath, 'Syntaxes', 'Makefile.plist'))).toBeTruthy();
-      expect(fs.existsSync(linkedRepoPath)).toBeTruthy();
-      return expect(fs.realpathSync(linkedRepoPath)).toBe(fs.realpathSync(repoPath));
+  describe("when the repository hasn't been cloned", () => {
+    it('clones the repository to ATOM_REPOS_HOME and links it to ATOM_HOME/dev/packages', () => {
+      const Develop = require('../lib/develop');
+      spyOn(Develop.prototype, 'getRepositoryUrl').andCallFake((packageName, callback) => {
+        const repoUrl = path.join(__dirname, 'fixtures', 'repo.git');
+        callback(null, repoUrl);
+      });
+      spyOn(Develop.prototype, 'installDependencies').andCallFake(function (packageDirectory, options) {
```

Comment:
```suggestion
      spyOn(Develop.prototype, 'installDependencies').andCallFake((packageDirectory, options) => {
```

Reply:
Uh oh, this is a breaking suggestion. `this` is dynamically bound in `function` literals but lexically bound in arrow literals so the latter is almost certainly faulty; there is no `this`˙introduced anywhere lexically.
It needs to stay with `function` to have any hope of working properly.

Reply:
Seems interesting it's causing a failure, as I haven't run into it before.

But since that was only a stylistic suggestion before, in that case I'll close the suggestion. Thanks for pursuing it!

Reply:
Well, maybe you also learned something, then. :P https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#cannot_be_used_as_methods